File: Problems/0001.-0100./0028._Find_the_Index_of_the_First_Occurence_in_a_String/FindtheIndexoftheFirstOccurenceinaString0.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        # Convert each to a list
        # lst1 = list(haystack)
        # lst2 = list(needle)

        # Create a variable to hold the length of the needle word
        n = len(needle)
        lst = []
        i = 0
        if needle in haystack:
            while i < len(haystack)- n and lst == []:
                if haystack[i:i+n] == needle:
                    lst.append(i)
                    break
                i += 1
            return i
        else:
            return -1

File: Problems/0001.-0100./0028._Find_the_Index_of_the_First_Occurence_in_a_String/FindtheIndexoftheFirstOccurenceinaString0.java
//!/bin/java

class Solution {
  public int strStr(String haystack, String needle) {
    // Variable to hold the length of the needle
    int m = needle.length();
    int[] ind = new int[]{-1};
    int i = 0;
    
    while (i < haystack.length()-m+1) {
      if ((haystack.substring(i,i+m)).equals(needle)) {
        ind[0] = i;
        break;
      }
      i += 1;
    }

    if (ind[0] >= 0) {
      return ind[0];
    } else {
      return ind[0];
    }
  }
}
